lib/web_of_science/query_author.rb

```diff
@@ -34,10 +34,46 @@ def uids
       # @return [Hash]
       def author_query
         params = queries.params_for_fields(empty_fields)
-        params[:queryParameters][:userQuery] = "AU=(#{names}) AND AD=(#{institution})"
+        compose_time_span(params)
+        update_user_query(params)
         params
       end
 
+      # Use options to limit the symbolic time span for harvesting publications; this limit applies
+      # to the dates publications are added or updated in WOS collections, not publication dates. To
+      # quote the API documentation:
+      #
+      #     The symbolicTimeSpan element defines a range of load dates. The load date is the date when a record
+      #     was added to a database. If symbolicTimeSpan is specified, the timeSpan parameter must be omitted.
+      #     If timeSpan and symbolicTimeSpan are both omitted, then the maximum publication date time span
+      #     will be inferred from the editions data.
+      #
+      #     The valid values are strings: '1week', '2week', '4week' (prior to today)
+      #
+      # @param [Hash] params from queries.params_for_fields
+      # @return [void]
+      def compose_time_span(params)
+        return if options[:update].blank?
+        # The Savon gem and WOS SOAP API are _very_ sensitive to the order of the params in the Hash; it must be
+        # reconstructed in the correct order - do not try conventional ruby Hash updates here.  This makes no sense
+        # to me [DLW], but learning from experience indicates the following works OK:
+        query = params[:queryParameters].dup
+        params[:queryParameters] = {
+          databaseId: query[:databaseId],
+          userQuery: query[:userQuery],
+          symbolicTimeSpan: '4week',
+          queryLanguage: query[:queryLanguage]
+        }
+      end
+
+      # @param [Hash] params from queries.params_for_fields
+      # @return [void]
+      def update_user_query(params)
+        params[:queryParameters][:userQuery] = "AU=(#{names}) AND AD=(#{institution})"
```

Comment:
Breaking this one-liner out to a separate method doesn't seem to add anything.  To me, there is less clarity because the model of "mutate a deep Hash" is not a good way to define behavior.  Ready to merge this otherwise.

Reply:
A possible refactor of all that is to encapsulate the query params into a class of it's own - most of it was coded as private details on the WOS-query, assuming all of it could remain private.  But, it has turned out that surfacing access to param details from the WOS-query has been useful.  A WOS-params class is a possibility, although punted so far.